requirements/requirements.in

```
@@ -31,7 +31,7 @@ billiard!=3.5.0.5  # Should be resolved in celery 4.3: https://github.com/celery
 mock==2.0.0
 Pillow>=6.2.2,<7.0.0
 phonenumberslite~=8.8
-psycopg2~=2.7.7
+psycopg2>=2.8.4  # Python 3.8 support
```

@kaapstorm I got an error when I tried to install this locally:
```
Error: pg_config executable not found.

pg_config is required to build psycopg2 from source.  Please add the directory
containing pg_config to the $PATH
```
Could we use the binary package instead? This one installs without a hitch, and much faster too because it does not need to be compiled.
```suggestion
psycopg2-binary>=2.8.4  # Python 3.8 support
```

---

I also ran into this and would be interested in `psycopg2-binary` if it'll install more smoothly.

---

Same! Agree with the suggested change, and was happy to see others have already made the suggestion I was also tentatively considering suggesting. Very validating!

---

I did consider this before approving the PR but according to the psycopg2 docs:

> Note The psycopg2-binary package is meant for beginners to start playing with Python and PostgreSQL without the need to meet the build requirements.
If you are the maintainer of a publish package depending on psycopg2 you shouldn’t use ‘psycopg2-binary’ as a module dependency. **For production use you are advised to use the source distribution.**

---

Here's the list of prerequisits for building from source: https://www.psycopg.org/docs/install.html#build-prerequisites

Perhaps we can put the binary package in `dev-requirements` and the source in `prod-requirements`?

---

Urgh. Sorry everyone. I didn't realise 2.8.4 had different requirements from 2.7.7.

Thanks for the link @snopoke. Using the binary in dev-requirements and the source in prod-requirements makes sense.

It seems we need to install `libpq-dev` (which, now that I look, was already installed on my machine). Installing `postgresql-common` also works for Ubuntu users.

[According to SO](https://stackoverflow.com/a/22868821), Mac users need to:

    $ brew install postgresql
    $ gem install pg

So just using the binary package in dev environments sounds simpler.

We could leave psycopg2 at 2.7.7 in prod-requirements until we're ready to use Python 3.8 in production.

---

Oof. I had not seen that warning that psycopg2-binary is a toy. Also, I use MacPorts instead of homebrew, so here's how I was able to install:
```
$ sudo port install postgresql96
$ sudo port select postgresql postgresql96
$ pip install 'psycopg2==2.8.5'
```

---

In case this helps someone else, I had to add the following line to my `.zshrc`
```
export LDFLAGS="-I/usr/local/opt/openssl/include -L/usr/local/opt/openssl/lib"
```
to get it to work (c.f. https://stackoverflow.com/questions/39767810/cant-install-psycopg2-package-through-pip-install-is-this-because-of-sierra)

---

Another data point: on mac OS Mojave, I ran into OpenSSL problems and ultimately fixed them by adding an older version of OpenSSL as described here: https://github.com/kelaberetiv/TagUI/issues/86#issuecomment-372045221

---

thank you @dannyroberts